Model/Export/Consumer.php

```diff
@@ -0,0 +1,36 @@
+<?php
+
+namespace StoreKeeper\StoreKeeper\Model\Export;
+
+use Psr\Log\LoggerInterface;
+use StoreKeeper\StoreKeeper\Model\ProductExporter;
+
+class Consumer
+{
+    private LoggerInterface $logger;
+    private ProductExporter $productExporter;
+
+    public function __construct(
+        LoggerInterface $logger,
+        ProductExporter $productExporter
+    ) {
+        $this->logger = $logger;
+        $this->productExporter = $productExporter;
+    }
+
+    /**
+     * @param string $request
+     * @return void
+     */
+    public function process($request): void
+    {
+        try {
+            $data = json_decode($request, true);
+            if ($data['entity'] == 'catalog_product') {
+                $this->productExporter->exportProductToCsv();
+            }
+        } catch (\Exception $e) {
+            $this->logger->error($e->getMessage());
```

Comment:
we should report the errors back to gui

Reply:
@ikulis Because it is processed in queue, we cannot trigger gui. In magento core export if something happened during export queue - result file just does not appears, and error recorded into log file

Reply:
can you at least log the trace?